bool save_updated_map
---
uint8 code
string message
